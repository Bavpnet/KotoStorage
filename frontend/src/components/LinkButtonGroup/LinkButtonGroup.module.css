@value primary from './../../globals.css';
@value mobile-max-width from './../../globals.css';

.links {
  display: flex;
  flex-direction: row;
  gap: 26px;
  justify-content: center;
  align-items: center;
  list-style: none;
  padding: 0;
}

.links.active {
  left: 0;
}

.links__btn {
  font-size: 1rem;
  font-weight: 400;
  transition: 0.3s;
}

.links__btn:hover {
  text-shadow: 0 0 1px black;
}

.line {
  display: none;
}

@media only screen and (max-width: mobile-max-width) {
  .links {
    position: fixed;
    top: 0;
    left: 100%;
    width: 100%;
    height: 100vh;
    display: flex;
    background: primary;
    flex-direction: column;
    justify-content: start;
    align-items: start;
    transition: 0.4s;
    padding-top: 64px;
    z-index: 5;
  }

  .links li {
    width: 100%;
  }

  .links__btn {
    margin-left: 22px;
  }

  .line {
    display: block;
    width: 100%;
    height: 1px;
    background-color: #101010;
  }
}
