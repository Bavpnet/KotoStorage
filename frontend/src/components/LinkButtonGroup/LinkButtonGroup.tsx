import { useNavigate } from 'react-router-dom'
import styles from './LinkButtonGroup.module.css'

type LinkButtonGroupProps = {
  isMenuOpen: boolean
  toggleMenu: (forceValue?: boolean) => void
}

export const LinkButtonGroup: React.FC<LinkButtonGroupProps> = ({
  isMenuOpen,
  toggleMenu
}) => {
  const linkButtons = ['About', 'Upload', 'Gallery']
  const navigate = useNavigate()

  const onLinkClick = (link: string) => {
    navigate(`/${link.toLowerCase()}`)
    toggleMenu(false)
  }

  return (
    <ul className={styles.links}>
      <li role="presentation" className={styles.line} />

      {linkButtons.map(link => (
        <li key={`linkbtn__${link}`}>
          <button
            className={styles.links__btn}
            onClick={() => onLinkClick(link)}
          >
            <span>{link}</span>
            {isMenuOpen && <span>{'->'}</span>}
          </button>
        </li>
      ))}

      <li role="presentation" className={styles.line} />
    </ul>
  )
}
