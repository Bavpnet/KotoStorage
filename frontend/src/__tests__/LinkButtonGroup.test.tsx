import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { LinkButtonGroup } from '../components/LinkButtonGroup'

describe('LinkButtonGroup', () => {
  it('renders correctly', () => {
    const toggleMenu = jest.fn()
    const tree = renderer
      .create(
        <MemoryRouter>
          <LinkButtonGroup isMenuOpen={false} toggleMenu={toggleMenu} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
