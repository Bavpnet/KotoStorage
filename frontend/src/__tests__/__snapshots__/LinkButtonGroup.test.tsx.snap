// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonGroup renders correctly 1`] = `
<ul
  className="links"
>
  <li
    className="line"
    role="presentation"
  />
  <li>
    <button
      className="linksBtn "
      onClick={[Function]}
    >
      <span>
        About
      </span>
    </button>
  </li>
  <li>
    <button
      className="linksBtn "
      onClick={[Function]}
    >
      <span>
        Upload
      </span>
    </button>
  </li>
  <li>
    <button
      className="linksBtn "
      onClick={[Function]}
    >
      <span>
        Gallery
      </span>
    </button>
  </li>
  <li
    className="line"
    role="presentation"
  />
</ul>
`;
